internal/mode/static/telemetry/platform.go

```diff
@@ -0,0 +1,103 @@
+package telemetry
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type k8sState struct {
+	node       v1.Node
+	namespaces v1.NamespaceList
+}
+
+type platformExtractor func(k8sState) (string, bool)
+
+func buildProviderIDExtractor(id string, platform string) platformExtractor {
+	return func(state k8sState) (string, bool) {
+		if strings.HasPrefix(state.node.Spec.ProviderID, id) {
+			return platform, true
+		}
+		return "", false
+	}
+}
+
+const (
+	gkeIdentifier       = "gce"
+	awsIdentifier       = "aws"
+	azureIdentifier     = "azure"
+	kindIdentifier      = "kind"
+	k3sIdentifier       = "k3s"
+	openshiftIdentifier = "node.openshift.io/os_id"
+	rancherIdentifier   = "cattle-system"
+
+	platformGKE       = "gke"
+	platformAWS       = "eks"
+	platformAzure     = "aks"
+	platformKind      = "kind"
+	platformK3S       = "k3s"
+	platformOpenShift = "openshift"
+	platformRancher   = "rancher"
+)
+
+var multiDistributionPlatformExtractors = []platformExtractor{
+	rancherExtractor,
+	openShiftExtractor,
+}
+
+var platformExtractors = []platformExtractor{
+	buildProviderIDExtractor(gkeIdentifier, platformGKE),
+	buildProviderIDExtractor(awsIdentifier, platformAWS),
+	buildProviderIDExtractor(azureIdentifier, platformAzure),
+	buildProviderIDExtractor(kindIdentifier, platformKind),
+	buildProviderIDExtractor(k3sIdentifier, platformK3S),
+}
+
+func getPlatform(node v1.Node, namespaces v1.NamespaceList) string {
+	state := k8sState{
+		node:       node,
+		namespaces: namespaces,
+	}
+
+	// must be run before providerIDPlatformExtractors as these platforms
+	// may have multiple platforms e.g. Rancher on K3S, and we want to record the
+	// higher level platform.
+	for _, extractor := range multiDistributionPlatformExtractors {
```

Comment:
we can simplify getPlatform by combining all extractors into one slice, so we only need a single loop and no extra ifs.
 
```go
var platformExtractors = []platformExtractor{
        openShiftExtractor, // must be run before providerIDPlatformExtractors
        rancherExtractor, //  must be run before providerIDPlatformExtractors
	buildProviderIDExtractor(gkeIdentifier, platformGKE),
	buildProviderIDExtractor(awsIdentifier, platformAWS),
	buildProviderIDExtractor(azureIdentifier, platformAzure),
	buildProviderIDExtractor(kindIdentifier, platformKind),
	buildProviderIDExtractor(k3sIdentifier, platformK3S),
	unknownProviderIDExtractor, // must run the last
}
```

Reply:
the slice (as opposed to a map) also has a specific order. You can also by mistake put the new code in a wrong place :)  - I assume that why in the current code, you also put clarifying comments.

you can limit the comments like below:
```go
var platformExtractors = []platformExtractor{
        openShiftExtractor, 
        rancherExtractor,
        // ID provider extractors must run after the rest
	buildProviderIDExtractor(gkeIdentifier, platformGKE),
	buildProviderIDExtractor(awsIdentifier, platformAWS),
	buildProviderIDExtractor(azureIdentifier, platformAzure),
	buildProviderIDExtractor(kindIdentifier, platformKind),
	buildProviderIDExtractor(k3sIdentifier, platformK3S),
	unknownProviderIDExtractor, 
}
```

the benefit here is that if you do need to add a new one, you just update the list, without changing getPlatform function.